### DISCUSSION TITLE
incorrect matching bracket highlighting

### DISCUSSION THREAD

**Discussion by dvic** [LazyVim/LazyVim#911] [Q&A]
It seems like this is happening in all languages, but if you have a bracket inside a string literal, the matching is incorrect, for example in lua:

![image](https://github.com/LazyVim/LazyVim/assets/1214337/528942e5-d4a9-451d-8db6-548e7c4b29c6)

Does anyone know how this can be fixed?

**Comment by dpetka2001** [comment]
No idea, I too experience it, but I think it has nothing to do with Lazyvim. Rather `mini.pairs` could be the issue, so you might be better off raising an issue over there to see if it can be solved.

**Reply by folke** [reply]
mini.pairs doesn't do any highlighting.

The highlights are coming from Neovim's matchparen, which doesn't do special treesitter stuff.

If you want more advanced parens matching, use something like `vim-matchup`

**Reply by dpetka2001** [reply]
Oh cool, nice to know @folke. Will try it out.
